Add RewriteController to allow easy handling of historic rewrites

nhp opened the pull request:
Added a new controller to be able to access the old rewrites vsf-api knows nothing about. It simply makes a DB call to core_url_rewrite, filtered by request_path and returns only the new final target.
## Usage
Simply called by /vsbridge/rewrite/target?request_path=old_url this returns {"code":200,"result":"target_path"}

As we agreed with Łukasz Romanowicz on slack we should bring this functionality to the main magento1 trunk.

sandermangel commented on magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/RewriteController.php:
```diff
@@ -0,0 +1,27 @@
+
+            $reader = Mage::getSingleton('core/resource')->getConnection('core_read');
+            $select = $reader->select()
+                             ->from('core_url_rewrite', ['target_path'])->where('request_path = ?', $requestPath);
```
Please add a `LIMIT 1` to the query to improve performance.

nhp replied:
I honestly thought that ->fetchOne() would do exactly like that but you get surprised by magento/Zend/Varien internals every now and then:) Limiting beforehand now introduced.

sandermangel replied:
I know, it really sucks :/
Thanks!

nhp commented:
Also added store_id handling as this could also be an issue and we added more specific result-status codes to reflect on what really was the problem instead of only returning 200 and especially 500.

sandermangel commented on magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/RewriteController.php:
```diff
@@ -0,0 +1,34 @@
+            $select = $reader->select()
+                             ->from('core_url_rewrite', ['target_path'])
+                             ->where('request_path = ?', $requestPath)
+                             ->where('store_id IN (?)', [Mage_Core_Model_App::ADMIN_STORE_ID, (int)Mage::app()->getStore()->getId()])
```
I've always tried to stay away from using the store id in the API, I've had issues with false matches on that.
might we require it to be set in the API call?

sandermangel commented on magento1-module/app/code/local/Divante/VueStorefrontBridge/controllers/RewriteController.php:
```diff
@@ -0,0 +1,34 @@
+            $select = $reader->select()
+                             ->from('core_url_rewrite', ['target_path'])
+                             ->where('request_path = ?', $requestPath)
+                             ->where('store_id IN (?)', [Mage_Core_Model_App::ADMIN_STORE_ID, (int)Mage::app()->getStore()->getId()])
```
I personally avoid using the globally set store id, had some issues in the past with having the wrong store.
might we require it to be set in the URL as param?

ResuBaka replied:
Here i think we should have both, the `Mage::app()->getStore()->getId()` as a fallback when no storeId is set in the request? 

Like for example:
```
$storeId = $this->getRequest()->getParam('storeId');
if (empty($storeId)) {
    $storeId = (int)Mage::app()->getStore()->getId();
}
```

What do you think about that? @sandermangel

nhp commented:
Do we still have open issues here?

nhp commented:
Any update on this one?

ResuBaka commented:
Here I have found a small Problem. I will Update the PR in the next days.

sandermangel commented:
@ResuBaka hi! any update on the PR changes?

ResuBaka commented:
I will update in the next days. 

The problem is that I need to send the request URLs html encoded.
